rd-kt/rd-framework/src/commonMain/kotlin/com/jetbrains/rd/framework/impl/RdTask.kt

```diff
@@ -228,22 +231,23 @@ class RdCall<TReq, TRes>(internal val requestSzr: ISerializer<TReq> = Polymorphi
     /**
      * Assigns a handler that executes the API asynchronously.
      */
-    override fun set(cancellationAndRequestScheduler: IScheduler?, handler: (Lifetime, TReq) -> RdTask<TRes>) {
+    override fun set(cancellationScheduler: IScheduler?, handlerScheduler: IScheduler?, handler: (Lifetime, TReq) -> RdTask<TRes>) {
         this.handler = handler
-        this.endpointSchedulerForHandlerAndCancellation = cancellationAndRequestScheduler
+        this.cancellationScheduler = cancellationScheduler
+        this.handlerScheduler = handlerScheduler
     }
 
-    constructor(cancellationAndRequestScheduler: IScheduler? = null, handler:(Lifetime, TReq) -> RdTask<TRes>) : this() { set(cancellationAndRequestScheduler, handler) }
+    constructor(cancellationScheduler: IScheduler? = null, handlerScheduler: IScheduler? = null, handler:(Lifetime, TReq) -> RdTask<TRes>) : this() { set(cancellationScheduler, handlerScheduler, handler) }
 
     /**
      * Assigns a handler that executes the API synchronously.
      */
-    constructor(cancellationAndRequestScheduler: IScheduler? = null, handler: (TReq) -> TRes) : this () { set(cancellationAndRequestScheduler, handler) }
+    constructor(cancellationScheduler: IScheduler? = null, handlerScheduler: IScheduler? = null, handler: (TReq) -> TRes) : this () { set(cancellationScheduler, handlerScheduler, handler) }
 
     override fun onWireReceived(buffer: AbstractBuffer) {
         val taskId = RdId.read(buffer)
 
-        val wiredTask = EndpointWiredRdTask(bindLifetime, this, taskId, endpointSchedulerForHandlerAndCancellation ?: wireScheduler)
+        val wiredTask = EndpointWiredRdTask(bindLifetime, this, taskId, cancellationScheduler ?: SynchronousScheduler)
```

Comment:
A couple of side notes:
- Yes, seems like `endpointSchedulerForHandlerAndCancellation` was only ever used for cancellation, and weren't actually used for _handler_, even if its name suggests otherwise. So, the corresponding API change is non-breakling one in practice.
- This is an (intended) behaviorial change: by default, `SynchronousScheduler` will be used for cancellation. We'll need to port it to C++ part, I guess.